internal/gitea/team.go

```diff
@@ -0,0 +1,149 @@
+package gitea
+
+import (
+	"fmt"
+	"log"
+	"slices"
+	"strings"
+
+	"code.gitea.io/sdk/gitea"
+	"github.com/jnsgruk/releasegen/internal/repos"
+)
+
+const giteaPerPage = 10
+
+// FetchOrgRepos creates a slice of RepoDetails types representing the repos
+// owned by the Gitea org.
+func FetchOrgRepos(org OrgConfig) ([]repos.RepoDetails, error) {
+	orgRepos := []repos.RepoDetails{}
+
+	gtClient, err := org.GiteaClient()
+	if err != nil {
+		return nil, fmt.Errorf("error creating gitea client: %s", err)
```

Comment:
Using the newer `%w` fmt verb is slightly better in cases like this, as it properly wraps the error so that the resulting `error` value's `Unwrap()` gives back the original error. See more here: https://pkg.go.dev/errors